Documentation/vmware-backdoor.md

```diff
@@ -0,0 +1,25 @@
+# VMware Backdoor #
+
+coreos-cloudinit is capable of reading userdata and metadata from the VMware
+backdoor. This datasource can be enable with the `--from-vmware-backdoor` flag.
+Userdata and metadata are passed from the hypervisor to the virtual machine
+through guest variables. The following guest variables and their expected types
+are supported by coreos-cloudinit:
+
+|            guest variable             |              type               |
+|:--------------------------------------|:--------------------------------|
+| `hostname`                            | `hostname`                      |
+| `interface.<n>.name`                  | `string`                        |
+| `interface.<n>.mac`                   | `MAC address`                   |
+| `interface.<n>.dhcp`                  | `{"yes", "no"}`                 |
+| `interface.<n>.role`                  | `{"public", "private"}`         |
+| `interface.<n>.ip.<m>.address`        | `CIDR IP address`               |
+| `interface.<n>.route.<l>.gateway`     | `IP address`                    |
+| `interface.<n>.route.<l>.destination` | `CIDR IP address`               |
+| `dns.<x>`                             | `IP address`                    |
```

Comment:
lets do `dns.server.<x>` since we may want `dns.domain` or `dns.search.<y>` in the future.